src/transformers/trainer.py

```diff
@@ -3716,3 +3720,45 @@ def _gather_and_numpify(self, tensors, name):
             tensors = distributed_concat(tensors)
 
         return nested_numpify(tensors)
+
+    def _add_sm_patterns_to_gitignore(
+        self, patterns: List[str] = [
+            "*.sagemaker-uploading",
+            "*.sagemaker-uploaded"
+        ]) -> None:
```

Comment:
It's weird to me to store the patterns (which won't really change) as a default argument. let's make a constant with them instead (and just have `self` here).

Reply:
Ok, I'll change it.